Up2dateService/Up2dateService/SetupManager/SetupManager.cs

```diff
@@ -264,15 +378,51 @@ private void RefreshPackageList()
                     updatedPackage.InstallDate = null;
                     updatedPackage.EstimatedSize = null;
                     updatedPackage.UrlInfoAbout = null;
-                    if (updatedPackage.Status != PackageStatus.Downloading && updatedPackage.Status != PackageStatus.Installing)
-                    {
+                    if (updatedPackage.Status != PackageStatus.Downloading &&
+                        updatedPackage.Status != PackageStatus.Installing)
                         updatedPackage.Status = PackageStatus.Downloaded;
-                    }
                 }
+
+                if (updatedPackage.Filepath.Contains(NugetExtension))
+                    using (var zipFile = ZipFile.OpenRead(updatedPackage.Filepath))
+                    {
+                        var nuspec = zipFile.Entries.First(zipArchiveEntry => zipArchiveEntry.Name.Contains(".nuspec"));
+                        using (var nuspecStream = nuspec.Open())
+                        {
+                            using (var sr = new StreamReader(nuspecStream, Encoding.UTF8))
+                            {
+                                var xmlData = sr.ReadToEnd();
+                                var doc = new XmlDocument();
+                                doc.LoadXml(xmlData);
+                                updatedPackage.DisplayName = doc.GetElementsByTagName("title")[0].InnerText;
+                                updatedPackage.ProductName = doc.GetElementsByTagName("id")[0].InnerText;
+                                updatedPackage.DisplayVersion = doc.GetElementsByTagName("version")[0].InnerText;
+                                updatedPackage.Publisher = doc.GetElementsByTagName("authors")[0].InnerText;
+                            }
+                        }
+
+                        using (var ps = PowerShell.Create())
+                        {
+                            const string psCommand = @"choco list -li";
+                            ps.AddScript(psCommand);
+                            var result = ps.Invoke<string>();
+                            if (result.Any(item => item.Contains(updatedPackage.ProductName)))
```

Comment:
This check doesn't seem reliable enough...
What if another product name just includes that name?
For instanse I can have a package named MyPython....